src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -637,15 +648,13 @@ func (b *ByocAws) getPrivateFqdn(fqn qualifiedName) string {
 	return fmt.Sprintf("%s.%s", safeFqn, b.PrivateDomain) // TODO: consider merging this with ServiceDNS
 }
 
-func (b *ByocAws) getProjectDomain(zone string) string {
+func (b *ByocAws) getProjectDomain(account, zone string) string {
 	if b.ProjectName == "" {
 		return "" // no project name => no custom domain
 	}
-	projectLabel := byoc.DnsSafeLabel(b.ProjectName)
-	if projectLabel == byoc.DnsSafeLabel(b.TenantID) {
-		return byoc.DnsSafe(zone) // the zone will already have the tenant ID
-	}
-	return projectLabel + "." + byoc.DnsSafe(zone)
+	h := sha256.New()
+	fmt.Fprintf(h, "%s.%s.%s.%s", account, b.ProjectName, b.TenantID, zone)
```

Comment:
to prepare for future identity providers, we should start using a GUID for users and use that in this hash (not the GitHub name)

Reply:
also needs Stack (even though it's always `beta` now)

Reply:
I was thinking keep tenant id as the main identifier, but extend it with domain, like edwardrf@gitlab, or edw$facebook